app/models/experimentation/notifications_experiment.rb

```diff
@@ -39,22 +41,45 @@ def self.eligible_patients
                                              .having("count(patient_id) > 1"))
     end
 
-    def enroll_patients(date)
+    def enroll_patients(date, limit = MAX_PATIENTS_PER_DAY)
       eligible_patients(date)
-        .limit(MAX_PATIENTS_PER_DAY)
+        .limit([remaining_enrollments_allowed(date), limit].min)
         .includes(:assigned_facility, :registration_facility, :medical_history)
         .includes(latest_scheduled_appointments: [:facility, :creation_facility])
-        .in_batches(of: ENROLLMENT_BATCH_SIZE)
+        .in_batches(of: MEMBERSHIPS_BATCH_SIZE)
         .each_record { |patient| random_treatment_group.enroll(patient, reporting_data(patient, date)) }
     end
 
     def monitor(date)
+      # Add check_notification_statuses in a follow up PR
       mark_visits
       evict_patients
     end
 
-    def send_notifications(date)
-      # Send notifications to memberships_to_notify(date)
+    def schedule_notifications(date)
+      reminder_templates.each do |template|
+        memberships_to_notify(template, date)
+          .in_batches(of: MEMBERSHIPS_BATCH_SIZE)
+          .each_record { |membership| schedule_notification(membership, template, date) }
+      end
+    end
+
+    def schedule_notification(membership, template, date)
+      return if Notification.find_by(
+        reminder_template: template,
+        experiment: self,
+        patient_id: membership.patient_id
+      )
+
+      Notification.create!(
+        experiment: self,
+        message: template.message,
+        patient_id: membership.patient_id,
+        purpose: :experimental_appointment_reminder,
+        remind_on: date,
+        reminder_template: template,
+        status: "pending"
+      )
```

Comment:
Is it not possible to use `find-or-create` here? https://guides.rubyonrails.org/active_record_querying.html#find-or-create-by